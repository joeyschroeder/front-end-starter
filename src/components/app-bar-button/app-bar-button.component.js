import PropTypes from 'prop-types';
import React from 'react';
import styles from './app-bar-button.style.scss';

export const AppBarButton = props => {

    const { children, onClick } = props;

    return (
        <button className={styles.root} type='button' onClick={onClick}>{children}</button>
    );
};

AppBarButton.propTypes = {
    children: PropTypes.oneOfType([ PropTypes.arrayOf(PropTypes.node), PropTypes.node ]),
    onClick: PropTypes.func
};

AppBarButton.defaultProps = {
    children: null,
    onClick: null
};
