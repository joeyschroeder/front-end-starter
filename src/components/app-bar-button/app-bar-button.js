import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import React from 'react';
import styles from './app-bar-button.style.scss';

export const AppBarButton = (props) => {
  const { children, onClick, to } = props;

  if (to)
    return (
      <Link className={styles.root} to={to}>
        {children}
      </Link>
    );

  return (
    <button className={styles.root} type="button" onClick={onClick}>
      {children}
    </button>
  );
};

AppBarButton.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  onClick: PropTypes.func,
  to: PropTypes.string,
};

AppBarButton.defaultProps = {
  children: null,
  onClick: null,
  to: '',
};
