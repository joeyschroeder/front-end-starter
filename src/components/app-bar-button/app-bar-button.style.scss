@import "../../styles/config/variables";

.root {
  border-right: $border-width solid $color-border;
  display: inline-block;
  padding: $spacer;
  text-decoration: none;
  text-transform: uppercase;
}
