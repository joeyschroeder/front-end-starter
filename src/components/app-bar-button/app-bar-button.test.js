import { AppBarButton } from './app-bar-button';
import React from 'react';
import { create } from 'react-test-renderer';

describe('<AppBarButton />', () => {
  it('should render correctly', () => {
    const component = create(<AppBarButton />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
