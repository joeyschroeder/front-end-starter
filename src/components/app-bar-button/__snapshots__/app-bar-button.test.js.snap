// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBarButton /> should render correctly 1`] = `
<button
  className="root"
  onClick={null}
  type="button"
/>
`;
